pandas/tests/indexes/multi/test_sorting.py

```diff
@@ -289,3 +290,20 @@ def test_sort_values_nan():
         levels=[["A", "B", "C"], ["D"]], codes=[[0, 1, 2], [-1, -1, 0]]
     )
     tm.assert_index_equal(result, expected)
+
+
+def test_sort_values_monotonic_incomparable():
+    mi = MultiIndex.from_arrays(
+        [
+            [1, Timestamp("2000-01-01")],
+            [3, 4],
+        ]
+    )
+
+    match = "'<' not supported between instances of 'Timestamp' and 'int'"
+    with pytest.raises(TypeError, match=match):
+        mi._sort_levels_monotonic(raise_if_incomparable=True)
```

Comment:
Ah sorry, I meant sort_values that shows the warning

Reply:
I updated the test to sort_values. However, note that sort_values raises a TypeError not a warning. The TypeError is the same behavior as exists on 1.5 and is also the same type error you get with `Index([1, Timestamp("2000-01-01")]).sort_values()`. Are you thinking of `MultiIndex._union` which catches the TypeError and converts it to a warning?

Reply:
Exactly, thx